server/text_generation_server/models/custom_modeling/flash_qwen2_modeling.py

```diff
@@ -306,12 +335,24 @@ def forward(
         max_s: int,
         true_max_s: int,
         prefill_cache_indices: Optional[torch.Tensor],
+        inputs_embeds: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
-        hidden_states = self.embed_tokens(input_ids)
+
+        # if inputs_embeds are supplied from an external model (vision model) then avoid embedding input_ids
+        if inputs_embeds is not None:
```

Comment:
No, remove this.

The way it's done here, is that we life input_ids to the parent class, and this always takes `input_embeds`. Makes signatures much cleaner. (It's already done for llama this way if you want to check)

Reply:
yes agreed this is much cleaner. I've updated the classes in the latest commit to follow the same pattern as llama